server/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -197,9 +193,11 @@ protected GeoShapeQueryBuilder newShapeQueryBuilder(String fieldName, Supplier<G
 
     @Override
     public Query buildShapeQuery(QueryShardContext context, MappedFieldType fieldType) {
-        if (fieldType.typeName().equals(GeoShapeFieldMapper.CONTENT_TYPE) == false) {
+        if (!validContentTypes().contains(fieldType.typeName())) {
```

Comment:
For readability we prefer `== false` rather than `!`

Reply:
OK, done